packages/build/src/plugins_core/edge_functions/get_edge_function_stats.ts

```diff
@@ -0,0 +1,16 @@
+import { promises as fs } from 'fs'
+import { join, resolve } from 'path'
+
+export const getEdgeFunctionStats = async ({ buildDir, constants: { EDGE_FUNCTIONS_DIST: distDirectory } }) => {
+  const edgeFunctionsDistPath = resolve(buildDir, distDirectory)
+  const manifestPath = join(edgeFunctionsDistPath, 'manifest.json')
+  const data = await fs.readFile(manifestPath)
+  // @ts-expect-error TypeScript is not aware that parse can handle Buffer
+  const manifestData = JSON.parse(data)
+  const numGenEfs = Object.values(manifestData.function_config).filter(
+    (config: { generator?: string }) => config.generator,
+  ).length
+  const totalNumEfs = manifestData.routes.length + manifestData.post_cache_routes.length
+
+  return { num_gen_efs: numGenEfs, total_num_efs: totalNumEfs }
```

Comment:
I think with these tags the cardinality could be huge, because it's basically adding infinite possible amount of data in the tags.

So tags should be always limited in scope like `os: windows, os: linux, os: mac`, but not `tag: <infinte-number>`
And for every new tag value a new metric is basically generated.

What could work are tags like: `functionType: generated, functionType: user`, but that would not give us the total amount of functions, but only the num of deploys that contain either/or.

I think a new distributed metric in datadog that gets the number of functions for generated and not generated might be better.